src/form/Form.jsx

```diff
@@ -692,27 +692,27 @@ function Form({
     const servar = field.servar;
     if (servar.repeat_trigger === 'set_value') {
       const defaultValue = getDefaultFieldValue(field);
-      const { value: previousValue } = getFieldValue(field, fieldValues);
+      const { value: previousValue, valueList } = getFieldValue(
+        field,
+        fieldValues
+      );
 
       // Add a repeated row if the value went from unset to set
-      // And this is the last field in a set of repeated fields
       const isPreviousValueDefaultArray =
         isEmptyArray(previousValue) && isEmptyArray(defaultValue);
 
-      if (
-        // index === repeatedRowCount - 1 &&
-        (previousValue === defaultValue || isPreviousValueDefaultArray) &&
-        value !== defaultValue
-      ) {
-        repeatRowOperation = 'add';
-      }
+      // And this is the last field in a set of repeated fields
+      const isLastRepeatedField =
+        Array.isArray(valueList) && index === valueList.length - 1;
```

Comment:
nit: i think if you check if valueList is truthy you can assume it's an array here. if it breaks it's an upstream error anyways